packages/compass-components/src/components/guide-cue/guide-cue.tsx

```diff
@@ -0,0 +1,222 @@
+import React, {
+  useEffect,
+  useCallback,
+  useRef,
+  useMemo,
+  useState,
+} from 'react';
+import { guideCueService, type ShowCueEventDetail } from './guide-cue-service';
+import { GuideCue as LGGuideCue } from '@leafygreen-ui/guide-cue';
+import type { GroupName } from './guide-cue-groups';
+import { css, cx } from '../..';
+import { rafraf } from '../../utils/rafraf';
+
+const hiddenPopoverStyles = css({
+  display: 'none !important',
+  opacity: '0 !important',
+  transition: 'none !important',
+});
+
+const getDataCueId = ({
+  cueId,
+  groupId,
+}: {
+  cueId: string;
+  groupId?: string;
+}): string => {
+  return `guide-cue-${cueId}${groupId ? `-${groupId}` : ''}`;
+};
+
+type LGGuideCueProps = React.ComponentProps<typeof LGGuideCue>;
+
+type GroupAndStep =
+  | {
+      groupId: GroupName;
+      step: number;
+    }
+  | {
+      groupId?: never;
+      step?: never;
+    };
+
+// omit the props we are handling
+export type GuideCueProps<T> = Omit<
+  LGGuideCueProps,
+  | 'currentStep'
+  | 'refEl'
+  | 'numberOfSteps'
+  | 'onDismiss'
+  | 'open'
+  | 'onPrimaryButtonClick'
```

Comment:
I was just trying to add a GuideCue for the signal popover and I think we might want these to be available props? In our case the pimary button says something like "See them in action" and should behave as if the insight was clicked